addons/l10n_in_pos/static/src/overrides/models/payment_screen.js

```diff
@@ -13,4 +15,41 @@ patch(PaymentScreen.prototype, {
         }
         return super.toggleIsToInvoice();
     },
+    async toggleShippingDatePicker() {
+        if (this.pos.company.country_id?.code === "IN") {
+            if (!this.currentOrder.getShippingDate()) {
+                this.dialog.add(ShipLater, {
+                    title: _t("Select the shipping date"),
+                    getPayload: (shippingDate, stateId) => {
+                        this.currentOrder.setShippingDate(shippingDate);
+                        this.currentOrder.setPlaceOfSupply(stateId);
```

Comment:
Remove this method and set the variable directly there.

```
this.currentOrder.update({l10n_in_state_id: state})
```
State needs to be the record itself not an id

Reply:
> Remove this method and set the variable directly there.
> 
> ```
> this.currentOrder.update({l10n_in_state_id: state})
> ```
> 
> State needs to be the record itself not an id

I cannot remove getPayload it is mandatory prop for `DatePickerPopup`.

Reply:
yes but do what I say inside of getPayload